superflore/generators/bitbake/yocto_recipe.py

```diff
@@ -51,11 +53,15 @@ def __init__(self, name, distro, src_uri, tar_dir):
         self.archive_name = None
         self.license_md5 = None
         self.tar_dir = tar_dir
-        self.downloadArchive()
-        self.src_sha256 = hashlib.sha256(
-            open(self.getArchiveName(), 'rb').read()).hexdigest()
-        self.src_md5 = hashlib.md5(
-            open(self.getArchiveName(), 'rb').read()).hexdigest()
+        if self.getArchiveName() not in md5_cache and \
```

Comment:
This should be an `or` in case one or the other is preinitialized but not the other.

Reply:
I'll add this in #105 and merge this, since you're mid-travel.